packages/simcore-sdk/src/simcore_sdk/node_ports_v2/nodeports_v2.py

```diff
@@ -27,6 +29,20 @@
 log = logging.getLogger(__name__)
 
 
+class OutputsCallbacks(ABC):
```

Comment:
Here I have a few comments:
1. this OutputsCallbacks, and actually set_multiple are only used by the dynamic sidecar, why are they coming with this >complex stuff inside the node ports package? Would it not make sense that you build on top instead of creating these weird constructs?
2. The tests you added do not check anything regarding how this callbacks work,
3. Sorry, but this looks like C++ man... ;)

Reply:
> Here I have a few comments:
> 
> 1. this OutputsCallbacks, and actually set_multiple are only used by the dynamic sidecar, why are they coming with this >complex stuff inside the node ports package? Would it not make sense that you build on top instead of creating these weird constructs?

This works on the same principle as the progress bar works. The quick answer is no. The implementation must leave here since it's the place where these events can be detected, similar to what's happening with the progress bar.

> 2. The tests you added do not check anything regarding how this callbacks work,

No, I've added some other tests which checks how the callbacks are being treated. I have also made this fully optional.

> 3. Sorry, but this looks like C++ man... ;)